Library/Client.ts

```diff
@@ -240,30 +240,30 @@ class Client {
     }
 
     public getEnvelope(
-        data: ContractsModule.Contracts.Data<ContractsModule.Contracts.Domain>,
-        tagOverrides?: { [key: string]: string; }) {
-        if (data && data.baseData) {
-            data.baseData.ver = 2;
-
-            // if no properties are specified just add the common ones
-            if (!data.baseData.properties) {
-                data.baseData.properties = this.commonProperties;
-            } else {
-                // otherwise, check each of the common ones
-                for (var name in this.commonProperties) {
-                    // only override if the property `name` has not been set on this item
-                    if (!data.baseData.properties[name]) {
-                        data.baseData.properties[name] = this.commonProperties[name];
+        data: Contracts.Data<Contracts.Domain>,
+        tagOverrides?: { [key: string]: string; }): Contracts.Envelope {
+        if (Contracts.domainSupportsProperties(data.baseData)) { // Do instanceof check. TS will automatically cast and allow the properties property
```

Comment:
This is slightly different logic, but I think it is more correct (plus type safe!). The difference is the previous logic would dump a default properties property on any supplied baseData. While all of our data types support this key, it is not on the base Domain class, so new ones may intentionally not have it